### PR TITLE
[tt-transformers] Update integration tests to adopt HF RoPE directly

### PR DESCRIPTION
**Pull request by epam-ioannis-alexiou**
### Ticket
Close https://github.com/tenstorrent/tt-metal/issues/32988

### Problem description
The Q and K attention weights are assigned to HF model computational graph to match tensor output for increased similarity between hidden states of each layer. This is temporary till exclusion of vision branch is implemented in convert_hf_to_meta_llama_format(). The exclusion is impleemted to the patching script needs to be removed.

### What's changed
The patching script for Q and K projection weights of attention is removed. Also the function to build attention mask is imported from common file. Also the intermediate states are pulled directly from HF model without the need to pass TT tensor in each layer of the transformer. 

### Checklist

- [x] [![[post-commit] all - Static checks, linters etc.](https://github.com/tenstorrent/tt-metal/actions/workflows/all-static-checks.yaml/badge.svg)](https://github.com/tenstorrent/tt-metal/actions/workflows/all-static-checks.yaml)

#### Model tests

- [x] [t3k_integrations_tests](https://github.com/tenstorrent/tt-metal/actions/runs/23240461717) relevant task is `t3k_llama3.2-90b-vision_tests`.

### CODE REVIEW

**Review by maksim-tsishkouski-epam — APPROVED** [review]
Can you please run [t3k integration tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-integration-tests.yaml) to check that this test is passing

File: models/tt_transformers/tt/multimodal/llama_cross_attention.py [hunk 1]
```diff
@@ -264,7 +264,14 @@ def forward_decode(
...
             )
-
+        if full_text_row_masked_out_mask_1NSH.dtype != output.dtype:
+            full_text_row_masked_out_mask_1NSH = ttnn.typecast(full_text_row_masked_out_mask_1NSH, output.dtype)
+        if full_text_row_masked_out_mask_1NSH.shape != output.shape:
+            full_text_row_masked_out_mask_1NSH = ttnn.reshape(
+                full_text_row_masked_out_mask_1NSH,
+                list(output.shape),
+                list(full_text_row_masked_out_mask_1NSH.padded_shape),
+            )
```
Comment (gwangTT) [inline models/tt_transformers/tt/multimodal/llama_cross_attention.py]: Please include a comment that explain why this fix was necessary

File: tests/pipeline_reorg/t3k_integration_tests.yaml [hunk 2]
```diff
@@ -37,32 +37,32 @@
...
+  model-name: llama3.2-vision
+
+- name: t3k_n300_mesh_llama3.2-vision_tests
+  cmd: run_t3000_spoof_n300_llama3.2-11b-vision_freq_tests
+  skus:
+    wh_llmbox:
+      timeout: 20
+  owner_id: U03FJB5TM5Y # Colman Glagovich
+  team: models
+  model-name: n300 mesh llama3.2-vision
```
Comment (gwangTT) [inline tests/pipeline_reorg/t3k_integration_tests.yaml]: from our discussion on Slack, 11B model does not require integration tests anymore. e2e tests are sufficient for tier-3 models (11B is one of them).

File: models/tt_transformers/tests/multimodal/test_llama_image_transformer.py [hunk 3]
```diff
@@ -167,13 +136,12 @@ def test_image_transformer_inference(batch, num_chunks, mesh_device, is_global):
         tens_input = ttnn.to_torch(attention_input, mesh_composer=ttnn.ConcatMeshToTensor(mesh_device, dim=0))[
             0, :, :, :
         ].reshape(batch * num_chunks, ntok + npadtt, dim)
-        feats = callable_reference(tens_input[:, :ntok, :], attention_mask=mask)
+
+        feats = callable_reference(tens_input[:, :ntok, :], attention_mask=mask, output_hidden_states=True)
         reference_output = feats.last_hidden_state
         if return_intermediate != None:
```
[the remark this answers is not in the document]
Comment (gwangTT) [reply models/tt_transformers/tests/multimodal/test_llama_image_transformer.py]: This is a good suggestion. Please adopt this.

File: models/tt_transformers/tests/multimodal/test_llama_image_transformer.py [hunk 4]
```diff
@@ -167,13 +136,12 @@ def test_image_transformer_inference(batch, num_chunks, mesh_device, is_global):
         tens_input = ttnn.to_torch(attention_input, mesh_composer=ttnn.ConcatMeshToTensor(mesh_device, dim=0))[
             0, :, :, :
         ].reshape(batch * num_chunks, ntok + npadtt, dim)
-        feats = callable_reference(tens_input[:, :ntok, :], attention_mask=mask)
+
+        feats = callable_reference(tens_input[:, :ntok, :], attention_mask=mask, output_hidden_states=True)
```
[the remark this answers is not in the document]
Comment (gwangTT) [reply models/tt_transformers/tests/multimodal/test_llama_image_transformer.py]: Seems like a good suggestion. Consider adopting?

File: models/tt_transformers/tests/multimodal/test_llama_image_transformer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (epam-ioannis-alexiou) [reply models/tt_transformers/tests/multimodal/test_llama_image_transformer.py]: done in [603e7c7](https://github.com/tenstorrent/tt-metal/pull/39547/commits/603e7c7049fe18c0bc2c58dc206ef0952356d3c6)

File: models/tt_transformers/tests/multimodal/test_llama_image_transformer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (epam-ioannis-alexiou) [reply models/tt_transformers/tests/multimodal/test_llama_image_transformer.py]: done in [603e7c7](https://github.com/tenstorrent/tt-metal/pull/39547/commits/603e7c7049fe18c0bc2c58dc206ef0952356d3c6)

File: models/tt_transformers/tt/multimodal/llama_cross_attention.py (hunk 1 above)
Comment (epam-ioannis-alexiou) [reply models/tt_transformers/tt/multimodal/llama_cross_attention.py]: The latest merge rendered this code unnecessary. The reason of its existence was because of the ttnn.mult failed to do broadcasting properly when one middle dimension was unequal.
For example it will fail to broadcast the mask to ouput (now it does not happen).
  - mask: [1, 1, 16, 128] BFLOAT4_B — n_local_heads=16 (so it's a 2-device N300 setup with 11B)
  - output: [1, 1, 32, 128] BF16 — xq's logical shape got expanded to the padded tile size (32)

File: tests/pipeline_reorg/t3k_integration_tests.yaml (hunk 2 above)
Comment (epam-ioannis-alexiou) [reply tests/pipeline_reorg/t3k_integration_tests.yaml]: Yes I left these for experimentation as 11b runs faster. I have left only 90b case for now, done in [fc5eb8c](https://github.com/tenstorrent/tt-metal/pull/39547/commits/fc5eb8c470e7a44ae8ae5575e6e01cebe700ce9d)

File: .github/workflows/t3000-integration-tests.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ on:
           - falcon40b
           # - llama3.2-vision # Tier 3 models
           # - n300 mesh llama3.2-vision # Tier 3 models
-          # - llama3.2-90b-vision # Tier 3 models
+          - llama3.2-90b-vision # Tier 3 models
```
Comment (gwangTT) [inline .github/workflows/t3000-integration-tests.yaml]:
```suggestion
          - llama3.2-90b-vision # Tier 2 model
```

**Review by gwangTT — APPROVED** [review]
LGTM. Thanks, Ioannis!

**Review by roseli-TT — APPROVED** [review]
